src/callbacks_step/analysis_dg3d_parallel.jl

```diff
@@ -74,8 +74,8 @@ function integrate_via_indices(func::Func, u,
     # Initialize integral with zeros of the right shape
     # Pass `zero(SVector{nvariables(equations), eltype(u))}` to `func` since `u` might be empty, if the
     # current rank has no elements, see also https://github.com/trixi-framework/Trixi.jl/issues/1096.
-    integral = zero(func(zero(SVector{nvariables(equations), eltype(u)}), 1, 1, 1, 1,
-                         equations, dg, args...))
+    integral = zero(func(zeros(nvariables(equations), nnodes(dg), nnodes(dg),
+                               nnodes(dg), 1), 1, 1, 1, 1, equations, dg, args...))
```

Comment:
I had to change the initialization of `integral` for it to work with the analysis routines for MHD. More specifically, `:l2_divb` and `:linf_divb` require a multi-dimensional array, not only an SVector of size `nvariables(equations)`. The original serial implementation works as well:
```julia
integral = zero(func(u, 1, 1, 1, 1, equations, dg, args...))
```
However, it was changed because it causes problems with idling ranks (see https://github.com/trixi-framework/Trixi.jl/issues/1096). 

Is there a better way to do this, @ranocha?

Reply:
We need at least to ensure we use the correct element type `eltype(u)`. I don't have a perfect solution for this. Thus, let's choose the working one that comes with additional allocations.

Reply:
Done in 8b1b179.